src/backend/distributed/commands/table.c

```diff
@@ -396,28 +396,7 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand)
 			 * creation without a name and allow foreign key creation with the mentioned
 			 * command.
 			 */
-			ColumnDef *columnDefinition = (ColumnDef *) command->def;
-			List *columnConstraints = columnDefinition->constraints;
-
-			Constraint *constraint = NULL;
-			foreach_ptr(constraint, columnConstraints)
-			{
-				if (constraint->contype == CONSTR_FOREIGN)
-				{
-					rightRelationId = RangeVarGetRelid(constraint->pktable, lockmode,
-													   alterTableStatement->missing_ok);
-
-					/*
-					 * Foreign constraint validations will be done in workers. If we do not
-					 * set this flag, PostgreSQL tries to do additional checking when we drop
-					 * to standard_ProcessUtility. standard_ProcessUtility tries to open new
-					 * connections to workers to verify foreign constraints while original
-					 * transaction is in process, which causes deadlock.
-					 */
-					constraint->skip_validation = true;
-					break;
-				}
-			}
+			SkipValidationIfCommandDefinesForeignKey((Node *) alterTableStatement);
```

Comment:
Aren't we already calling `SkipValidationIfCommandDefinesForeignKey ` at the top of this function?